src/librustc_lint/bad_style.rs

```diff
@@ -288,11 +288,8 @@ impl LateLintPass for NonSnakeCase {
     }
 
     fn check_pat(&mut self, cx: &LateContext, p: &hir::Pat) {
-        if let &PatKind::Binding(_, ref path1, _) = &p.node {
-            // Exclude parameter names from foreign functions (they have no `Def`)
```

Comment:
But they were going about it the wrong way... I suppose I can make this *actually* look for foreign `fn` items so the intent is preserved.